src/responder/kcm/secrets/secrets.c

```diff
@@ -214,13 +380,20 @@ static int local_db_check_peruid_number_of_secrets(TALLOC_CTX *mem_ctx,
     }
 
     if (res->count >= req->quota->max_uid_secrets) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "Cannot store any more secrets for this client (basedn %s) "
-              "as the maximum allowed limit (%d) has been reached\n",
-              ldb_dn_get_linearized(cli_basedn),
-              req->quota->max_uid_secrets);
-        ret = ERR_SEC_INVALID_TOO_MANY_SECRETS;
-        goto done;
+        /* We reached the limit. Let's try to removed the
+         * oldest expired credential to free some space. */
+        ret = local_db_remove_oldest_expired_secret(res, req);
+        if (ret != EOK) {
+            if (ret == ERR_NO_MATCHING_CREDS) {
+                DEBUG(SSSDBG_OP_FAILURE,
+                      "Cannot store any more secrets for this client (basedn %s) "
+                      "as the maximum allowed limit (%d) has been reached\n",
+                      ldb_dn_get_linearized(cli_basedn),
+                      req->quota->max_uid_secrets);
```

Comment:
Hi,

this is not strictly related to your patch. While testing I got irritated because this log message was showing a different number than I configured with `max_uid_ccaches` option. The reason is that the value is unconditionally incremented by `2` in `ccdb_secdb_init()`. I'm not sure, what would be best here, leave it as it is, reduce the output by `2` or say in the message that some internals are included in the limit?

bye,
Sumit

Reply:
Fixed